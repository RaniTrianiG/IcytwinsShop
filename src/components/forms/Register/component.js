import React from 'react';
import { Field } from 'redux-form';
import { View } from 'react-native';
import { Button } from 'react-native-elements';
import PropTypes from 'prop-types';
import InputField from '../../elements/TextInput';
import style from './styles';

const Component = props => {
  const { handleSubmit } = props;
  return (
    <View>
      <Field component={InputField} name="name" placeholder="Nama" label="Nama" />
      <Field component={InputField} name="email" placeholder="Email" label="Email" />
      <Field component={InputField} name="phoneNum" placeholder="Nomor Telepon" label="Nomor Telepon" />
      <Field component={InputField} name="password" placeholder="Password" label="Password" secureTextEntry />
      <Field
        component={InputField}
        name="confirmPass"
        placeholder="Ulangi Password"
        label="Ulangi Password"
        secureTextEntry
      />
      <View style={style.buttonFooter}>
        <Button containerStyle={{ flex: 1, margin: 2 }} onPress={handleSubmit} title="Daftar" />
      </View>
    </View>
  );
};

Component.propTypes = {
  handleSubmit: PropTypes.func.isRequired
};

export default Component;
