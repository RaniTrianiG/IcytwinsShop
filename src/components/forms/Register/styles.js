import { StyleSheet } from 'react-native';

const style = StyleSheet.create({
  buttonFooter: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between'
  },
  checkboxContainer: {
    alignItems: 'center',
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginBottom: 15
  }
});

export default style;
