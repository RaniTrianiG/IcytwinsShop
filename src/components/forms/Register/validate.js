import validateInput from '../../../utils/validateInput';

const validate = values => {
  const { name, email, phoneNum, password, confirmPass } = values;
  const errors = {
    name: validateInput(name, [
      {
        rule: 'required',
        message: 'PIN tidak boleh kosong'
      }
    ]),
    email: validateInput(email, [
      {
        rule: 'required',
        message: 'PIN tidak boleh kosong'
      }
    ]),
    phoneNum: validateInput(phoneNum, [
      {
        rule: 'required',
        message: 'PIN tidak boleh kosong'
      }
    ]),
    password: validateInput(password, [
      {
        rule: 'required',
        message: 'PIN tidak boleh kosong'
      }
    ]),
    confirmPass: validateInput(confirmPass, [
      {
        rule: 'required',
        message: 'PIN tidak boleh kosong'
      }
    ])
  };
  return errors;
};

export default validate;
