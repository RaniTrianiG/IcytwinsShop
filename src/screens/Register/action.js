/* eslint-disable import/prefer-default-export */
import { ACTIONS, API } from '../../constants';
import { AXIOS } from '../../configs';
import { reqOption } from '../../utils';

export const postRegister = (data, successCallback) => async dispatch => {
  const option = await reqOption();
  dispatch({ type: ACTIONS.POST_REGISTER_START });
  AXIOS.post(API.REGISTER, data, option)
    .then(resp => {
      dispatch({ type: ACTIONS.POST_REGISTER_SUCCESS, data: resp });
      successCallback();
    })
    .catch(error => {
      dispatch({ type: ACTIONS.POST_REGISTER_FAILED, error });
    });
};
