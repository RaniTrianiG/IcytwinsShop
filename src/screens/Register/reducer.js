import { ACTIONS } from '../../constants';
import { objectUpdater } from '../../utils';

const initialState = {
  isLoading: false,
  error: null,
  data: null
};

const reducers = (state = initialState, action) => {
  const { error, data } = action;
  switch (action.type) {
    case ACTIONS.POST_REGISTER_START:
      return objectUpdater(state, { isLoading: true });
    case ACTIONS.POST_REGISTER_SUCCESS:
      return objectUpdater(state, { isLoading: false, data });
    case ACTIONS.POST_REGISTER_FAILED:
      return objectUpdater(state, { isLoading: false, error });
    default:
      return state;
  }
};

export default reducers;
