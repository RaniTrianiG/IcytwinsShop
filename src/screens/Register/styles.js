import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  logoContainer: {
    alignItems: 'center',
    height: 250
  },
  modalContainer: {
    backgroundColor: 'white'
  },
  screenContainer: {
    backgroundColor: 'white',
    padding: 10
  }
});

export default styles;
