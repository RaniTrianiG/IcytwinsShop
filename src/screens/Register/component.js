import React from 'react';
import { View, ScrollView, Text } from 'react-native';
import PropTypes from 'prop-types';
import Modal from 'react-native-modal';
import FormRegister from '../../components/forms/Register';
import styles from './styles';
import Logo from '../../assets/svg/login-illustration.svg';

class Register extends React.Component {
  state = {
    isModalVisible: false
  };

  _handleSubmit = async data => {
    const { actions } = this.props;
    actions.postRegister(data);
  };

  _handleModal = () => {
    const { isModalVisible } = this.state;
    this.setState({ isModalVisible: !isModalVisible });
  };

  render() {
    const { isModalVisible } = this.state;
    return (
      <ScrollView contentContainerStyle={styles.screenContainer}>
        <View style={styles.logoContainer}>
          <Logo width="100%" height="100%" />
        </View>
        <FormRegister onSubmit={this._handleSubmit} />
        <Modal isVisible={isModalVisible} onBackdropPress={this._handleModal}>
          <View style={styles.modalContainer}>
            <Text>Success Modal</Text>
          </View>
        </Modal>
      </ScrollView>
    );
  }
}

Register.defaultProps = {
  actions: {}
};

Register.propTypes = {
  actions: PropTypes.object
};

export default Register;
