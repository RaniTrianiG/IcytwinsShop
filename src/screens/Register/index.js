import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as actions from './action';
import Component from './component';

const mapStateToProps = state => {
  return {
    ...state.register
  };
};

const mapDispatchToProps = dispatch => {
  return {
    actions: bindActionCreators(actions, dispatch)
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Component);
