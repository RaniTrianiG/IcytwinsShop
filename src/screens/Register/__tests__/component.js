import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { reducer as formReducer } from 'redux-form';

import renderer from 'react-test-renderer';
import Register from '../component';

jest.mock('@react-native-community/netinfo');
jest.mock('react-native-navigation');

describe('Register Tests', () => {
  const mockStore = configureMockStore([thunk]);
  let initialState;
  let store;
  let props;
  let wrapper;
  beforeEach(() => {
    props = {
      actions: {
        postRegister: jest.fn()
      },
      isLoading: false
    };
    initialState = {
      form: formReducer
    };
    wrapper = shallow(<Register {...props} />);
  });

  afterEach(() => {
    store.clearActions();
  });

  it('renders correctly with initial state', () => {
    store = mockStore(initialState);
    const tree = renderer
      .create(
        <Provider store={store}>
          <Register {...props} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Simulate Form Submission', () => {
    const spy = jest.spyOn(wrapper.instance(), '_handleSubmit');
    wrapper.instance().forceUpdate();
    wrapper.find('ReduxForm').simulate('submit');
    expect(spy).toBeCalled();
  });

  it('Simulate modal visibility', () => {
    const spy = jest.spyOn(wrapper.instance(), '_handleModal');
    wrapper.instance().forceUpdate();
    wrapper.find('ReactNativeModal').simulate('backdropPress');
    expect(spy).toBeCalled();
  });

  it('renders correctly with Modal state', () => {
    wrapper.setState({ isModalVisible: true });
    expect(wrapper).toMatchSnapshot();
  });
});
