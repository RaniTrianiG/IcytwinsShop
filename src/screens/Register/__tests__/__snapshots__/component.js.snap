// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Tests renders correctly with Modal state 1`] = `
<ScrollViewMock
  contentContainerStyle={
    Object {
      "backgroundColor": "white",
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 250,
      }
    }
  >
    <SvgMock
      height="100%"
      width="100%"
    />
  </View>
  <ReduxForm
    destroyOnUnmount={true}
    enableReinitialize={false}
    forceUnregisterOnUnmount={false}
    form="register"
    getFormState={[Function]}
    keepDirtyOnReinitialize={false}
    onSubmit={[Function]}
    persistentSubmitErrors={false}
    pure={true}
    shouldAsyncValidate={[Function]}
    shouldError={[Function]}
    shouldValidate={[Function]}
    shouldWarn={[Function]}
    submitAsSideEffect={false}
    touchOnBlur={true}
    touchOnChange={false}
    updateUnregisteredFields={false}
    validate={[Function]}
  />
  <ReactNativeModal
    animationIn="slideInUp"
    animationInTiming={300}
    animationOut="slideOutDown"
    animationOutTiming={300}
    avoidKeyboard={false}
    backdropColor="black"
    backdropOpacity={0.7}
    backdropTransitionInTiming={300}
    backdropTransitionOutTiming={300}
    coverScreen={true}
    customBackdrop={null}
    deviceHeight={null}
    deviceWidth={null}
    hasBackdrop={true}
    hideModalContentWhileAnimating={false}
    isVisible={true}
    onBackButtonPress={[Function]}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalShow={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    propagateSwipe={false}
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    useNativeDriver={false}
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
        }
      }
    >
      <Text>
        Success Modal
      </Text>
    </View>
  </ReactNativeModal>
</ScrollViewMock>
`;

exports[`Register Tests renders correctly with initial state 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "backgroundColor": "white",
      "padding": 10,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 250,
        }
      }
    >
      <SvgMock
        height="100%"
        width="100%"
      />
    </View>
    <View>
      <View
        style={
          Object {
            "marginBottom": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "marginBottom": 10,
              "marginLeft": 10,
            }
          }
        >
          Nama
        </Text>
        <View
          style={
            Object {
              "paddingHorizontal": 10,
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderBottomWidth": 1,
                "borderColor": "#ccc",
                "borderRadius": 5,
                "borderWidth": 1,
                "flexDirection": "row",
                "marginBottom": 5,
                "padding": 3,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              editable={true}
              name="name"
              onBlur={[Function]}
              onChangeText={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              placeholder="Nama"
              rejectResponderTermination={true}
              style={
                Object {
                  "alignSelf": "center",
                  "color": "black",
                  "flex": 1,
                  "fontSize": 18,
                  "minHeight": 40,
                }
              }
              testID="RNE__Input__text-input"
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "marginBottom": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "marginBottom": 10,
              "marginLeft": 10,
            }
          }
        >
          Email
        </Text>
        <View
          style={
            Object {
              "paddingHorizontal": 10,
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderBottomWidth": 1,
                "borderColor": "#ccc",
                "borderRadius": 5,
                "borderWidth": 1,
                "flexDirection": "row",
                "marginBottom": 5,
                "padding": 3,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              editable={true}
              name="email"
              onBlur={[Function]}
              onChangeText={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              placeholder="Email"
              rejectResponderTermination={true}
              style={
                Object {
                  "alignSelf": "center",
                  "color": "black",
                  "flex": 1,
                  "fontSize": 18,
                  "minHeight": 40,
                }
              }
              testID="RNE__Input__text-input"
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "marginBottom": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "marginBottom": 10,
              "marginLeft": 10,
            }
          }
        >
          Nomor Telepon
        </Text>
        <View
          style={
            Object {
              "paddingHorizontal": 10,
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderBottomWidth": 1,
                "borderColor": "#ccc",
                "borderRadius": 5,
                "borderWidth": 1,
                "flexDirection": "row",
                "marginBottom": 5,
                "padding": 3,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              editable={true}
              name="phoneNum"
              onBlur={[Function]}
              onChangeText={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              placeholder="Nomor Telepon"
              rejectResponderTermination={true}
              style={
                Object {
                  "alignSelf": "center",
                  "color": "black",
                  "flex": 1,
                  "fontSize": 18,
                  "minHeight": 40,
                }
              }
              testID="RNE__Input__text-input"
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "marginBottom": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "marginBottom": 10,
              "marginLeft": 10,
            }
          }
        >
          Password
        </Text>
        <View
          style={
            Object {
              "paddingHorizontal": 10,
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderBottomWidth": 1,
                "borderColor": "#ccc",
                "borderRadius": 5,
                "borderWidth": 1,
                "flexDirection": "row",
                "marginBottom": 5,
                "padding": 3,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              editable={true}
              name="password"
              onBlur={[Function]}
              onChangeText={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              placeholder="Password"
              rejectResponderTermination={true}
              secureTextEntry={true}
              style={
                Object {
                  "alignSelf": "center",
                  "color": "black",
                  "flex": 1,
                  "fontSize": 18,
                  "minHeight": 40,
                }
              }
              testID="RNE__Input__text-input"
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "marginBottom": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "marginBottom": 10,
              "marginLeft": 10,
            }
          }
        >
          Ulangi Password
        </Text>
        <View
          style={
            Object {
              "paddingHorizontal": 10,
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderBottomWidth": 1,
                "borderColor": "#ccc",
                "borderRadius": 5,
                "borderWidth": 1,
                "flexDirection": "row",
                "marginBottom": 5,
                "padding": 3,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              editable={true}
              name="confirmPass"
              onBlur={[Function]}
              onChangeText={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              placeholder="Ulangi Password"
              rejectResponderTermination={true}
              secureTextEntry={true}
              style={
                Object {
                  "alignSelf": "center",
                  "color": "black",
                  "flex": 1,
                  "fontSize": 18,
                  "minHeight": 40,
                }
              }
              testID="RNE__Input__text-input"
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <View
          style={
            Object {
              "borderRadius": 3,
              "flex": 1,
              "margin": 2,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#2089dc",
                  "borderColor": "#2089dc",
                  "borderRadius": 3,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 18,
                    "paddingBottom": 1,
                    "paddingTop": 2,
                    "textAlign": "center",
                  }
                }
              >
                Daftar
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <Modal
      animationType="none"
      hardwareAccelerated={false}
      hideModalContentWhileAnimating={false}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      onRequestClose={[Function]}
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
      transparent={true}
      visible={false}
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "black",
            "bottom": 0,
            "height": 1334,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 750,
          }
        }
      />
      <View
        hideModalContentWhileAnimating={false}
        onModalHide={[Function]}
        onModalWillHide={[Function]}
        onModalWillShow={[Function]}
        pointerEvents="box-none"
        scrollHorizontal={false}
        scrollOffset={0}
        scrollOffsetMax={0}
        scrollTo={null}
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
            "margin": 37.5,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          }
        }
        supportedOrientations={
          Array [
            "portrait",
            "landscape",
          ]
        }
        swipeThreshold={100}
      >
        <View
          style={
            Object {
              "backgroundColor": "white",
            }
          }
        >
          <Text>
            Success Modal
          </Text>
        </View>
      </View>
    </Modal>
  </View>
</RCTScrollView>
`;
